File: core/src/com/blakephillips/game/ecs/components/HungerComponent.java
package com.blakephillips.game.ecs.components;

import com.badlogic.ashley.core.Component;

public class HungerComponent implements Component {
    private int hungerValue;

    public int getHunger() {
        return hungerValue;
    }

    public void setHungerValue(int hungerValue) {
        if (hungerValue > 100) {
            this.hungerValue = 100;
            return;
        }

        this.hungerValue = hungerValue;
    }

}

File: core/src/com/blakephillips/game/ecs/components/HungerComponent.kt
package com.blakephillips.game.ecs.components

import com.badlogic.ashley.core.Component

class HungerComponent : Component {
    var hunger = 0
        set(value) {
            if (value > 100) {
                field = 100
                return
            }
            field = value
        }
}
